internal/functions/serve/serve.go

```diff
@@ -216,3 +221,134 @@ func Run(ctx context.Context, slug string, envFilePath string, noVerifyJWT *bool
 	fmt.Println("Stopped serving " + utils.Bold(localFuncDir))
 	return nil
 }
+
+func runServeAll(ctx context.Context, envFilePath string, noVerifyJWT *bool, importMapPath string, fsys afero.Fs) error {
+	// 1. Sanity checks.
+	{
+		if err := utils.LoadConfigFS(fsys); err != nil {
+			return err
+		}
+		if err := utils.AssertSupabaseDbIsRunning(); err != nil {
+			return err
+		}
+		if envFilePath != "" {
+			if _, err := fsys.Stat(envFilePath); err != nil {
+				return fmt.Errorf("Failed to read env file: %w", err)
+			}
+		}
+		if importMapPath != "" {
+			// skip
+		} else if f, err := fsys.Stat(utils.FallbackImportMapPath); err == nil && !f.IsDir() {
+			importMapPath = utils.FallbackImportMapPath
+		}
+		if importMapPath != "" {
+			if _, err := fsys.Stat(importMapPath); err != nil {
+				return fmt.Errorf("Failed to read import map: %w", err)
+			}
+		}
+	}
+
+	// 2. Parse user defined env
+	userEnv, err := ParseEnvFile(envFilePath)
+	if err != nil {
+		return err
+	}
+
+	// 3. Start container
+	{
+		_ = utils.Docker.ContainerRemove(ctx, utils.DenoRelayId, types.ContainerRemoveOptions{
+			RemoveVolumes: true,
+			Force:         true,
+		})
+
+		env := []string{
+			"JWT_SECRET=" + utils.JWTSecret,
+			"SUPABASE_URL=http://" + utils.KongId + ":8000",
+			"SUPABASE_ANON_KEY=" + utils.AnonKey,
+			"SUPABASE_SERVICE_ROLE_KEY=" + utils.ServiceRoleKey,
+			"SUPABASE_DB_URL=postgresql://postgres:postgres@localhost:" + strconv.FormatUint(uint64(utils.Config.Db.Port), 10) + "/postgres",
+		}
+		verifyJWTEnv := "VERIFY_JWT=true"
+		if noVerifyJWT != nil {
+			verifyJWTEnv = "VERIFY_JWT=false"
+		}
+		env = append(env, verifyJWTEnv)
+
+		cwd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+
+		binds := []string{filepath.Join(cwd, utils.FunctionsDir) + ":" + relayFuncDir + ":ro,z"}
+		// If a import map path is explcitly provided, mount it as a separate file
+		if importMapPath != "" {
+			binds = append(binds, filepath.Join(cwd, importMapPath)+":"+customDockerImportMapPath+":ro,z")
+		}
+
+		// bind deno cache directory
+		cachePath, err := utils.GetDenoCachePath()
+		if err != nil {
+			return err
+		}
+		binds = append(binds, cachePath+":/root/.cache/deno:rw,z")
```

Comment:
Does deno cache work consistently across host and container OS? ie. if my host is windows, and the cache is mounted into a linux container, would it work?

Reply:
We provide a container-specific cache (defaults to `~/.supabase/deno_cache`). This is intended to be used only by the container and acts independent to user's local Deno installation.

Reply:
ic, it sounds like this cache folder doesn't need to be exposed to the host OS then. Wdyt about using docker named volumes to manage it? We started using it for backing up the `PGDATA` directory between cli restarts. Example here https://github.com/supabase/cli/blob/main/internal/db/start/start.go#L77 and https://github.com/supabase/cli/blob/main/internal/stop/stop.go#L52

Reply:
yep, sounds like that's a good option

Reply:
Updated in [c5b3291](https://github.com/supabase/cli/pull/828/commits/c5b32917b0fdc35041161119d1fbd78f79b76f1e) `functions serve` forcefully removes the container and its volumes. So didn't add to remove volumes in `supabase stop`.